rir/src/ir/CodeStream.h

```diff
@@ -230,6 +230,26 @@ class CodeStream {
         sources.back() = idx;
     }
 
+    unsigned currentPos() const {
+        return pos;
+    }
+
+    unsigned currentSourcesSize() const {
+        return sources.size();
+    }
+
+    void remove(unsigned pc, unsigned size, unsigned sourcesIdx) {
+
+        for (unsigned i = 0; i < size; ++i) {
+            *reinterpret_cast<Opcode*>(&(*code)[pc + i]) = Opcode::nop_;
+            // patchpoints are fixed by just removing the binding to label
+            patchpoints.erase(pc + i);
+        }
+
+        // need to insert source slots for the nops occupying the immediate places
+        sources.insert(sources.begin() + sourcesIdx + 1, size - 1, 0);
```

Comment:
Well assuming that every insertion of a BC adds one slot, then if you knew how many instructions are before the pc, you would get sourceIdx?
So you could advance from 0 until you got to pc?